client/state/sharing/services/selectors.js

```diff
@@ -9,6 +19,58 @@ export function getKeyringServices( state ) {
 }
 
 /**
+ * Returns an object of service objects with the specified type.
+ *
+ * @param  {Object} state Global state tree
+ * @param  {String} type  Type of service. 'publicize' or 'other'.
+ * @return {Array}        Keyring services, if known.
+ */
+export function getKeyringServicesByType( state, type ) {
+	return filter( getKeyringServices( state ), { type: type } );
+}
+
+/**
+ * Returns an object of eligible service objects with the specified type.
+ *
+ * @param  {Object} state Global state tree
+ * @param  {Object} site  Site object.
+ * @param  {String} type  Type of service. 'publicize' or 'other'.
+ * @return {Array}        Keyring services, if known.
+ */
+export function getEligibleKeyringServices( state, site, type ) {
+	const services = getKeyringServicesByType( state, type );
+
+	if ( ! site ) {
```

Comment:
Instead of a site object, we could call to [`isJetpackSite`](https://github.com/Automattic/wp-calypso/blob/08b32a64728927babb7b093ba988e695c37eaafd/client/state/sites/selectors.js#L119-L134) directly. If it returns null we know the site isn't known.